fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/ValueIndexScrubbingToolsMissing.java

```diff
@@ -0,0 +1,177 @@
+import com.apple.foundationdb.record.provider.foundationdb.IndexMaintainer;
+import com.apple.foundationdb.record.provider.foundationdb.IndexScrubbingTools;
+import com.apple.foundationdb.record.query.plan.RecordQueryPlanner;
+import com.apple.foundationdb.record.query.plan.synthetic.SyntheticRecordFromStoredRecordPlan;
+import com.apple.foundationdb.record.query.plan.synthetic.SyntheticRecordPlanner;
+import com.apple.foundationdb.tuple.Tuple;
+import com.google.protobuf.Message;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+
+/**
+ * Index Scrubbing Toolbox for a Value index maintainer. Scrub missing value index entries.
+ */
+public class ValueIndexScrubbingToolsMissing implements IndexScrubbingTools<FDBStoredRecord<Message>> {
+    private Collection<RecordType> recordTypes = null;
+    private Index index;
+    private boolean allowRepair;
+    private boolean isSynthetic;
+
+    @Override
+    public void presetCommonParams(Index index, boolean allowRepair, boolean isSynthetic, Collection<RecordType> types) {
+        this.recordTypes = types;
+        this.index = index;
+        this.allowRepair = allowRepair;
+        this.isSynthetic = isSynthetic;
+    }
+
+    @Override
+    public String getName() {
+        return "Scrub missing value index entries";
+    }
+
+    @Override
+    public RecordCursor<FDBStoredRecord<Message>> getIterator(final TupleRange tupleRange, final FDBRecordStore store, int limit) {
+        final IsolationLevel isolationLevel = IsolationLevel.SNAPSHOT;
+        final ExecuteProperties.Builder executeProperties = ExecuteProperties.newBuilder()
+                .setIsolationLevel(isolationLevel)
+                .setReturnedRowLimit(limit);
+
+        final ScanProperties scanProperties = new ScanProperties(executeProperties.build(), false);
+        return store.scanRecords(tupleRange, null, scanProperties);
+    }
+
+    @Override
+    public Tuple getContinuation(final RecordCursorResult<FDBStoredRecord<Message>> result) {
+        final FDBStoredRecord<Message> storedRecord = result.get();
+        return storedRecord == null ? null : storedRecord.getPrimaryKey();
+    }
+
+    @Override
+    @Nullable
+    public CompletableFuture<Issue> handleOneItem(FDBRecordStore store, Transaction transaction,  final RecordCursorResult<FDBStoredRecord<Message>> result) {
+        if (recordTypes == null || index == null) {
+            throw new InternalError("presetParams was not called appropriately for this scrubbing tool");
+        }
+
+        final FDBStoredRecord<Message> rec = result.get();
+        if (rec == null || !recordTypes.contains(rec.getRecordType())) {
+            return CompletableFuture.completedFuture(null);
+        }
```

Comment:
These are very good points! 
The scenario above will be caught if the other index (of the new type) will be scrubbed. However, it will be much more efficient to scan the record space once and check every index that supports this verification. This is somehow similar to multi target indexing.

Reply:
Reading your question again - if the other index type is not indexed, and the current index entry still exists, it will not be caught.